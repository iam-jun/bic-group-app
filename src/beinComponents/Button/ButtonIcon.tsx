import React from 'react';
import {StyleProp, ViewStyle} from 'react-native';
import {StyleSheet, View} from 'react-native';
import {ExtendedTheme, useTheme} from '@react-navigation/native';

import {IconType} from '~/resources/icons';

import ButtonWrapper, {ButtonWrapperProps} from './ButtonWrapper';
import Text from '~/beinComponents/Text';
import spacing from '~/theme/spacing';

export interface ButtonIconProps extends ButtonWrapperProps {
  style?: StyleProp<ViewStyle>;
  iconWrapperStyle?: StyleProp<ViewStyle>;
  icon: IconType;
  tintColor?: string;
  label?: string;
}

const ButtonIcon: React.FC<ButtonIconProps> = ({
  style,
  icon,
  label,
  tintColor,
  iconWrapperStyle,
  ...props
}: ButtonIconProps) => {
  const theme: ExtendedTheme = useTheme();
  const styles = createStyle(theme);

  return (
    <View testID="button_icon" style={[styles.container, style]}>
      <ButtonWrapper
        style={[styles.icon, iconWrapperStyle]}
        leftIcon={icon}
        leftIconProps={{icon, tintColor}}
        {...props}
      />
      {label && (
        <Text.ButtonS testID="button_icon.label" style={styles.label}>
          {label}
        </Text.ButtonS>
      )}
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const {colors} = theme;
  return StyleSheet.create({
    container: {
      alignItems: 'center',
    },
    icon: {
      backgroundColor: colors.violet1,
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: spacing.borderRadius.small,
      width: 36,
      height: 36,
    },
    label: {
      marginTop: spacing.margin.small,
      textAlign: 'center',
    },
  });
};

export default ButtonIcon;
