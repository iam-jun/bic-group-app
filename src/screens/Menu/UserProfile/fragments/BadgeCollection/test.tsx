import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
