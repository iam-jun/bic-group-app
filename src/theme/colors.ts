export default {
  dark: {
    colors: {
      transparent: 'rgba(255,255,255,0)',
      primary: '#5D2C9A',
      primary1: '#F8F5FF',
      primary2: '#EADFFE',
      primary3: '#DBCAFD',
      primary4: '#BC9FF8',
      primary5: '#A17AF2',
      primary6: '#7843DD',
      primary7: '#421187',

      background: '#20272F',
      bgSecondary: '#2F3945',
      bgFocus: '#505E71',
      bgHover: '#3F4B5B',
      bgDisable: '#505E71',
      bgButtonPrimary: '#8B5CE9',
      bgButtonSecondary: '#20272E',
      BorderButtonSecondary: '#8B5CE9',
      borderFocus: '#627286',
      borderDisable: '#627286',
      borderCard: '#B2BDCD',
      borderDivider: '#3F4B5B',
      textInput: '#F9FAFB',
      placeholder: '#3F4B5B',
      surface: '#2F3945', //Neutral1 background color for elements containing content, such as cards.
      iconTint: '#D5DCE5',
      iconTintLight: '#77879C',
      iconTintReversed: '#2F3945',

      link: '#2396FF',
      activeLink: '#5B22B8',
      success: '#0AC458',
      error: '#E8455F',
      warning: '#FFBC1F',
      statusOrange: '#E98A32',

      textPrimary: '#F9FAFB',
      textSecondary: '#77879C',
      textSuccess: '#FFFFFF',
      textDanger: '#FFFFFF',
      textWarning: '#2F3945',
      textInfo: '#007AFF',
      textDisabled: '#627286',
      textReversed: '#2F3945',
      textTertiary: '#627286',

      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',

      //react-native-paper default field
      accent: '#6600cc', //secondary color for your app which complements the primary color.
      text: '#29272A', //text color for content.
      disabled: 'rgba(0, 0, 0, 0.26)', //color for disabled elements.
      backdrop: 'rgba(0, 0, 0, 0.5)', //color for backdrops of various components such as modals.
      onSurface: '#000000', //background color for snackbars
      notification: '#f50057', //background color for badges

      //TabBar
      tabBarBackground: '#2F3945', //Neutral0
      activeColor: '#8B5CE9', //Primary50
      inactiveColor: '#77879C', //Neutral50

      // ViewLabel
      ViewLabel: {
        color: '#999999',
        colorHeading: '#000000',
      },
    },
  },
  light: {
    colors: {
      transparent: 'rgba(255,255,255,0)',
      primary: '#5D2C9A', //primary is brand color
      primary1: '#F8F5FF',
      primary2: '#EADFFE',
      primary3: '#DBCAFD',
      primary4: '#BC9FF8',
      primary5: '#A17AF2',
      primary6: '#7843DD', //Main Brand
      primary7: '#421187', //Primary90

      background: '#FFFFFF', //Neutral0
      bgSecondary: '#F9FAFB', //Neutral1
      bgFocus: '#D5DCE5', //Neutral10
      bgHover: '#EAEDF2', //Neutral5
      bgDisable: '#D5DCE5', //Neutral10
      bgButtonPrimary: '#7843DD ', //Primary50
      bgButtonSecondary: '#FFFFFF', //Neutral0'
      BorderButtonSecondary: '#7843DD',
      borderFocus: '#B2BDCD', //Neutral20
      borderDisable: '#B2BDCD', //Neutral20
      borderCard: '#B2BDCD', //Neutral20
      borderDivider: '#EAEDF2', //`Neutral5
      textInput: '#2F3945', // Neutral80
      placeholder: '#EAEDF2', //Neutral5
      surface: '#F9FAFB', //Neutral1 background color for elements containing content, such as cards.
      iconTint: '#2F3945', //Neutral80
      iconTintLight: '#627286', //Neutral50
      iconTintReversed: '#FFFFFF', //Neutral0

      link: '#007AFF', //Blue50
      activeLink: '#421187', //Primary90
      success: '#00A64D', //Green50
      error: '#DD2C3E', //Red50
      warning: '#FFBC1F', //Yellow40
      statusOrange: '#E98A32', //Orange40

      textPrimary: '#2F3945', //Neutral80
      textSecondary: '#627286', //Neutral50
      textSuccess: '#FFFFFF', //Neutral0
      textDanger: '#FFFFFF', //Neutral0
      textWarning: '#2F3945', //Neutral80
      textInfo: '#007AFF', //Blue50
      textDisabled: '#B2BDCD', //Neutral20
      textReversed: '#FFFFFF', //Neutral0
      textTertiary: '#627286',

      facebook: '#2D88FF',
      google: '#FF2828',
      apple: '#FFFFFF',

      //react-native-paper default field
      accent: '#6600cc', //secondary color for your app which complements the primary color.
      text: '#29272A', //text color for content.
      disabled: 'rgba(0, 0, 0, 0.26)', //color for disabled elements.
      backdrop: 'rgba(0, 0, 0, 0.5)', //color for backdrops of various components such as modals.
      onSurface: '#000000', //background color for snackbars
      notification: '#f50057', //background color for badges

      //TabBar
      tabBarBackground: '#FFFFFF', //Neutral0
      activeColor: '#7843DD', //Primary50
      inactiveColor: '#627286', //Neutral50

      // ViewLabel
      ViewLabel: {
        color: '#999999',
        colorHeading: '#000000',
      },
    },
  },
};
