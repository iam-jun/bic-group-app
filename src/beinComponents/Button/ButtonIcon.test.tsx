import React from 'react';

import {renderWithRedux, cleanup, fireEvent} from '~/test/testUtils';
import colors from '~/theme/theme';

import ButtonIcon from './ButtonIcon';

afterEach(cleanup);

describe('ButtonDanger conponent', () => {
  const children = 'Button Icon';

  const testID = 'button_icon.test';

  const icon = 'iconSend';

  const label = 'BUTTON ICON';

  it('renders correctly', () => {
    const rendered = renderWithRedux(
      <ButtonIcon icon={icon} testID={testID} />,
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('renders label correctly', () => {
    const rendered = renderWithRedux(
      <ButtonIcon icon={icon} label={label} testID={testID} />,
    );
    const labelComponent = rendered.getByTestId('button_icon.label');

    expect(labelComponent).toBeDefined();
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders style correctly', () => {
    const {getByTestId} = renderWithRedux(
      <ButtonIcon icon={icon} style={{margin: 12}} testID={testID} />,
    );

    const component = getByTestId('button_icon');

    expect(component.props.style[1].margin).toBe(12);
  });

  it('renders iconWrapperStyle correctly', () => {
    const rendered = renderWithRedux(
      <ButtonIcon
        icon={icon}
        iconWrapperStyle={{backgroundColor: 'red'}}
        testID={testID}
      />,
    );

    const component = rendered.getByTestId(testID);

    expect(component.props.style.backgroundColor).toBe('red');
  });
});
