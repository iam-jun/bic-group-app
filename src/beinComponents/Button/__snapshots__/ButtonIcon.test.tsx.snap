// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDanger conponent renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
  testID="button_icon"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": 4,
        "height": 36,
        "justifyContent": "center",
        "opacity": 1,
        "width": 36,
      }
    }
    testID="button_icon.test"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="button_wrapper.content"
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "marginHorizontal": 8,
              },
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View
              testID="button_wrapper.icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ButtonDanger conponent renders label correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
  testID="button_icon"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": 4,
        "height": 36,
        "justifyContent": "center",
        "opacity": 1,
        "width": 36,
      }
    }
    testID="button_icon.test"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="button_wrapper.content"
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        hitSlop={
          Object {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "marginHorizontal": 8,
              },
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        >
          <View
            style={
              Array [
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <View
              testID="button_wrapper.icon"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": undefined,
          "fontFamily": "BeVietnamPro-Medium",
          "fontSize": 13,
        },
        Object {},
        Object {
          "marginTop": 8,
          "textAlign": "center",
        },
      ]
    }
    testID="button_icon.label"
  >
    BUTTON ICON
  </Text>
</View>
`;
